### ISSUE TITLE
  ERROR: Failed building wheel for turbodbc on OSx using pip

### ISSUE THREAD

**Issue by larkz** [blue-yonder/turbodbc#286]
Despite having installed `pybind11`, when using pip to install turbodbc, it throws the following error
```
ERROR: Command errored out with exit status 1:
   command: /Users/lliu/code/geo_region_split/.grs-env3.8/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/m7/g7sbjz3s2313tbv8xnz07xbhhxwz5f/T/pip-install-7zijgwi1/turbodbc/setup.py'"'"'; __file__='"'"'/private/var/folders/m7/g7sbjz3s2313tbv8xnz07xbhhxwz5f/T/pip-install-7zijgwi1/turbodbc/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/m7/g7sbjz3s2313tbv8xnz07xbhhxwz5f/T/pip-wheel-6hnxsqss --python-tag cp38
       cwd: /private/var/folders/m7/g7sbjz3s2313tbv8xnz07xbhhxwz5f/T/pip-install-7zijgwi1/turbodbc/
  Complete output (70 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.15-x86_64-3.8
  creating build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/options.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/connect.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/__init__.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/data_types.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/api_constants.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/connection.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/exceptions.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/constructors.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  copying turbodbc/cursor.py -> build/lib.macosx-10.15-x86_64-3.8/turbodbc
  running egg_info
  writing turbodbc.egg-info/PKG-INFO
  writing dependency_links to turbodbc.egg-info/dependency_links.txt
  writing requirements to turbodbc.egg-info/requires.txt
  writing top-level names to turbodbc.egg-info/top_level.txt
  reading manifest file 'turbodbc.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'turbodbc.egg-info/SOURCES.txt'
  running build_ext
  building 'libturbodbc' extension
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/private/var/folders/m7/g7sbjz3s2313tbv8xnz07xbhhxwz5f/T/pip-install-7zijgwi1/turbodbc/setup.py", line 207, in <module>
      setup(name = 'turbodbc',
    File "/Users/lliu/code/geo_region_split/.grs-env3.8/lib/python3.8/site-packages/setuptools/__init__.py", line 145, in setup
      return distutils.core.setup(**attrs)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/Users/lliu/code/geo_region_split/.grs-env3.8/lib/python3.8/site-packages/wheel/bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/Users/lliu/code/geo_region_split/.grs-env3.8/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 84, in run
      _build_ext.run(self)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/command/build_ext.py", line 340, in run
      self.build_extensions()
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/command/build_ext.py", line 449, in build_extensions
      self._build_extensions_serial()
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/command/build_ext.py", line 474, in _build_extensions_serial
      self.build_extension(ext)
    File "/Users/lliu/code/geo_region_split/.grs-env3.8/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 205, in build_extension
      _build_ext.build_extension(self, ext)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/command/build_ext.py", line 528, in build_extension
      objects = self.compiler.compile(sources,
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/ccompiler.py", line 565, in compile
      self._setup_compile(output_dir, macros, include_dirs, sources,
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/ccompiler.py", line 341, in _setup_compile
      pp_opts = gen_preprocess_options(macros, incdirs)
    File "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/ccompiler.py", line 1076, in gen_preprocess_options
      pp_opts.append("-I%s" % dir)
    File "/private/var/folders/m7/g7sbjz3s2313tbv8xnz07xbhhxwz5f/T/pip-install-7zijgwi1/turbodbc/setup.py", line 70, in __str__
      import pybind11
  ModuleNotFoundError: No module named 'pybind11'
  ----------------------------------------
  ERROR: Failed building wheel for turbodbc
```
Will this be addressed in the future?

**Comment by xhochy** [comment]
Please post `pip list` and the full `pip install turbodbc` log. This seems to be only a partial log.

**Comment by larkz** [comment]
I tried to install `turbodbc` on a fresh pip env, and it was successful.

The problem was that I had a custom library to relies on `turbodbc`, and initially it was giving problems in the .whl creation. 

It has been resolved when I simply added the `pybind11` lib requirement into the `setup_requires` section on my `setup.py`. My lib was able to install `pybind11` before installing `turbodbc` thereby resolving the issue.

```
setup_requires=[
        'pybind11'
    ]
```
